core/common/src/main/java/alluxio/concurrent/jsr/ForkJoinPool.java

```diff
@@ -2495,6 +2496,45 @@ public ForkJoinWorkerThread run() {
     }
   }
 
+  /**
+   * Default ForkJoinWorkerThreadFactory implementation; creates a new ForkJoinWorkerThread using
+   * the system class loader as the thread context class loader.
+   */
+  public static final class AlluxioForkJoinWorkerThreadFactory
+          implements ForkJoinWorkerThreadFactory {
+    private static final AccessControlContext ACC = contextWithPermissions(
+            // new RuntimePermission("setContextClassLoader"), // java9-concurrent-backport changed
+            new RuntimePermission("getClassLoader"));
+    // Thread id counter.
+    private static final AtomicLong sThreadId = new AtomicLong(0);
```

Comment:
minor question, do thread ids typically start from 0 or 1?

Reply:
That's how guava thread-factory that we use assign indices to threads.

Reply:
changed the name to better reflect what it's an index.

Reply:
ok, sounds good